src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs

```diff
@@ -517,7 +517,7 @@ public void SaveContextInfo(HostService hostService, BuildPhase phase)
                 {
                     continue;
                 }
-                using (var writer = new StreamWriter(Path.Combine(Environment.ExpandEnvironmentVariables(BaseDir), stepInfo.ContextInfoFile)))
```

Comment:
Is append the expected behavior from the very beginning? Move using writer outside of the `foreach` loop instead?